### ISSUE TITLE
HelmChart kots kind should auto detect Helm version

### ISSUE THREAD

**Issue by aj-jester** [replicatedhq/kots#865]
I was able to repro the Helm v2 vs. v3 issue. Below are two identical charts, only difference is the dependencies in v2 is in a separate requirements.yaml while the v3 one is in Chart.yaml, no requirements.yaml. What’s weird is in v3, even when values.yaml (inside the Helm Chart) is hardcoded to false the subchart still gets deployed, but when I do helm template --debug . it doesn’t show the resources from the subchart, but when I deploy it via KOTS it seems to be totally ignore and the subchart is deployed.

Both the V2 and V3 charts are in the ZIP file.

[toocoolforskool-0.0.1_helm.zip](https://github.com/replicatedhq/kots/files/4985094/toocoolforskool-0.0.1_helm.zip)

```
apiVersion: kots.io/v1beta1
kind: HelmChart
metadata:
  name: toocoolforskool
spec:
  chart:
    name: toocoolforskool
    chartVersion: 0.0.1
  values:
    favorite:
      drink: badam

  optionalValues:
    - when: '{{repl ConfigOptionEquals "toocool_sub" "1" }}'
      values:
        toocool_sub:
          enabled: true
```

**Comment by marccampbell** [comment]
@aj-jester could you try this with `helmVersion` set to `v3` in this file? By default, KOTS renders with the v2 helm library, and we require opting into the v3 rendering methods today. https://kots.io/reference/v1beta1/helmchart/

**Comment by glentakahashi** [comment]
ah @marccampbell that did work, although i encountered this error: https://github.com/replicatedhq/kots/issues/881 that i worked around temporarily by splitting every single resource into its own file. That's a bit tedious and verbose, but for now its a fine workaround. Would prefer to go back to the multiple resource setup though

**Comment by AmberAlston** [comment]
This is now available in v1.43.0 https://kots.io/release-notes/#1430
